### ISSUE TITLE
DataProcessor doesn't process JS-dates

### ISSUE THREAD

**Issue by sassomedia** [amcharts/amcharts5#22]
There seems to be a bug in the `am5.DataProcessor`. According to the [docs](https://www.amcharts.com/docs/v5/concepts/data/#Parsing_dates), any dates not in timestamp format should be parsed with the data processor:
```javascript
series.data.processor = am5.DataProcessor.new(root, { dateFields: ["date"] });
series.data.setAll(data);
```

❌ If the data contains dates in `Date` format, the dates are not parsed and the chart renders empty. CodePen of the issue here: https://codepen.io/sassomedia/pen/VwWgMNG?editors=0010

✅ If the data however contains dates in `String` format, the parsing works as expected with the additional `dateFormat` prop:
```javascript
series.data.processor = am5.DataProcessor.new(root, {
  dateFields: ["date"],
  dateFormat: "yyyy-MM-dd",
});
series.data.setAll(data);
```

**Comment by martynasma** [comment]
Fixed in [5.0.0-beta.29](https://github.com/amcharts/amcharts5/releases/tag/5.0.0-beta.29).

## [5.0.0-beta.29] - 2021-10-01

### Added
- Wherever date or number format can be set (`DateFormatter`, `NumberFormatter`, `DateAxis`, `ValueAxis`, `Exporting`) it can now be set as `Intl.DateTimeFormatOptions` (for date format) or `Intl.NumberFormatOptions` (for number format) object in additional to string-based format. [More info](https://www.amcharts.com/docs/v5/tutorials/formatting-date-time-and-numbers-using-intl-object/).
- New list type `ListAutoDispose` added.

### Changed
- `XYCursor` will now ignore mouse movement if chart is obstructed by other element, unless zoomin/panning.
- `series`, `xAxes`, and `yAxes` on an `XYChart` are now `ListAutoDispose`, which means that removing items from those lists will automatically dispose them. Set `autoDispose = false` on those lists if you are planning on removing and reusing items.

### Fixed
- `Sunburst` nodes were shown on drill-down when they were not supposed to in some cases.
- `Hierarchy` node elements were ignoring `templateField`.
- It was impossible to move `XYChart` scrollbars to some other container.
- Fixed memory leaking when scrolling `LineSeries`, and when disposing series or other items.
- `XYCursor` was not firing `selectended` event if `behavior` was set to `"selectX"`, `"selectY"`, or `"selectXY"`.
- `XYCursor` was hiding selection when when behavior was set to `"selectX"`, `"selectY"`, or `"selectXY"`.
- `SerialChart.series.clear()` was not working properly.

[Full change log](https://github.com/amcharts/amcharts5/blob/master/packages/shared/CHANGELOG.md).

[Download options](https://www.amcharts.com/download-v5/).

Make sure you clear your browser cache after upgrading. And feel free to contact us again if you are still experiencing this issue.

**Comment by sassomedia** [comment]
Fixed, thanks!
